### PR TITLE
DM-44754: Initial version of APDB status technote

### PR DESCRIPTION
**Pull request by andy-slac**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -4,9 +4,215 @@ Current status of APDB and PPDB implementation
...
+It is likely that the ``SSSource`` table is not needed for AP pipeline queries, in that case APDB does not need to include it.
+Some update operations in Cassandra cannot be implemented trivially and may need additional tables to support those updates.
+
+There are two possible options for performing daily updates:
+
+- send all updates to APDB first and then propagate them to PPDB via the regular replication process,
+- store all updates in PPDB, then query PPDB and propagate them to APDB.
+
+There are pros and cons for each of these approaches.
+In the first case APDB becomes the only authoritative source for all information and PPDB is only updated by a single agent-replication process.
```
Comment (timj) [inline index.rst]: I wasn't entirely sure what "agent - replication" meant here so wasn't sure whether it was "agent-replication" or "single agent replication".

File: index.rst [hunk 2]
```diff
@@ -4,9 +4,215 @@ Current status of APDB and PPDB implementation
...
+
+APDB's main purpose is to serve AP pipeline queries in the most efficient way.
+To satisfy that goal the schema of the main tables used by AP (``DiaObject``, ``DiaSource``, ``DiaForcedSource``) has been optimized:
+
+- partitioning and clustering of these tables is defined to match the queries used by AP pipeline tasks,
+- a special table was added to the schema to keep the latest version of ``DiaObject`` records.
+
+To avoid unconstrained growth of data in ``DiaSource`` and ``DiaForcedSource`` tables we plan periodic cleanups of the data outside the 12-month window.
+
+APDB also serves as a source of data for downstream PPDB, but the optimized schema of the main tables cannot serve the types of queries that are used for copies of the recent data.
```
Comment (timj) [inline index.rst:97]: I wasn't sure about "copy" vs "copies"

**Review by timj — APPROVED** [review]
I made some tweaks and asked a couple of clarifying questions. You can merge after you correct my corrections...

File: index.rst (hunk 1 above)
Comment (andy-slac) [reply index.rst]: It's "single agent" which is "replication process", I'll try to improve.

File: index.rst (hunk 2 above)
Comment (andy-slac) [reply index.rst:97]: Either one looks OK to me.

File: index.rst (hunk 1 above)
Comment (timj) [reply index.rst]: Maybe dropping the hyphen is enough, or else "a single replication agent process"

Comment (andy-slac) [reply index.rst]: I dropped "replication process", it should be clear what that agent is from the above.

**Pull request merged**